packages/engine/src/HedronEngine/HedronEngine.ts

```diff
@@ -15,11 +16,49 @@ export class HedronEngine {
   private store: EngineStore
   private sketchesUrl: string | null = null
   private sketchManager: SketchManager
+  private midi: Midi
 
   constructor() {
     this.store = createEngineStore()
     this.sketchManager = new SketchManager()
     this.renderer = new Renderer()
+    this.midi = new Midi()
+    this.midi.onMidiMessage.add((event) => {
+      const id = `${event.device.name} - ${event.message.data?.[1]}`.replace(/ /g, '_')
```

Comment:
This form of ID feels bad, just something quick for testing. Likely want it to also be generated as some kind of UUID on the MIDIEvent class.

Noticed there is a device.id, but unsure if that remains consistent between boots/usb port used. Though being able to use 2 of the same controller and have them mapped automattically, it's a feature! >.<